### PR TITLE
feat: re-implement fetch-python-requirements...

### PR DESCRIPTION
**Pull request by phaer**
in python. This canonicalizes package name with the official `packaging` package and handels TLS ca certificates, so that pip does not need --trusted-host args anymore.

### CODE REVIEW

File: nix/pkgs/fetchPythonRequirements/fetch-python-requirements.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            "HOME": HOME
+        }
+    )
+    return proc
+
+
+def wait_for_proxy(proxy_port, cafile):
+    timeout = time.time() + 60 * 5
+    req = urllib.request.Request('https://pypi.org')
+    req.set_proxy(f'127.0.0.1:{proxy_port}', 'http')
```
Comment (DavHau) [inline nix/pkgs/fetchPythonRequirements/fetch-python-requirements.py:59]: Do we still need the `http` variant, as we have set up ssl now?

**Review by DavHau — COMMENTED** [review]
Beautiful

File: nix/pkgs/fetchPythonRequirements/fetch-python-requirements.py (hunk 1 above)
Comment (phaer) [reply nix/pkgs/fetchPythonRequirements/fetch-python-requirements.py:59]: Good question! I just assumed it wouldn't hurt as we want to proxy all pypi requests anyways. In practice it does not seem to work without that line, haven't looked into details yet - I would assume it should.

Comment (phaer) [reply nix/pkgs/fetchPythonRequirements/fetch-python-requirements.py:59]: As discussed, we'll just leave this as-is for now

**Pull request merged**